### ISSUE TITLE
`HostedUIASWebAuthenticationSession.showHostedUI` never calls the continuation when `aswebAuthenticationSession.canStart` is false

### ISSUE THREAD

**Issue by ahavamorse** [aws-amplify/amplify-swift#3678]
### Describe the bug

When `canStart` is false a warning is printed to the console because the continuation is never called.
`SWIFT TASK CONTINUATION MISUSE: showHostedUI(url:callbackScheme:inPrivate:presentationAnchor:) leaked its continuation!`
This is caused by the change in #3466 to only start the session if `canStart` is true. The resulting behavior is that the web UI is never presented and the continuation is never called. If the session is not started, the continuation must be called

The `canStart` variable was only temporarily false (due to an animation) and if we waited a short amount of time it would resolve within half a second. We therefore fixed it by adding a wait and then checking canStart again after a short pause. We check several times and then time out after a few seconds. We either start the session or else call the continuation with an error. This way the continuation is always called and the app is never left hanging.

### Steps To Reproduce

```swift
Steps to reproduce the behavior:
1. Call the `Amplify.Auth.signInWithWebUI` method and successfully sign in
2. Without restarting the app call `Amplify.Auth.signOut()` (which will never return)
```

### Expected behavior

If the session will not be started an error should be returned that says `canStart` is false (In our case this was because the presentation anchor was not yet presented)

### Amplify Framework Version

2.28.0

### Amplify Categories

Auth

### Dependency manager

Swift PM

### Swift version

5.9

### CLI version

n/a

### Xcode version

15.0

### Relevant log output

```shell
SWIFT TASK CONTINUATION MISUSE: showHostedUI(url:callbackScheme:inPrivate:presentationAnchor:) leaked its continuation!
```

### Is this a regression?

No

### Platforms

iOS

### OS Version

iOS 17.4.1

### Device

iPhone SE (2nd Generation)

**Comment by ruisebas** [comment]
Hi @ahavamorse, thanks for opening this issue.

The team will take a look and post any updates here.

**Comment by phantumcode** [comment]
Issue fixed and released in version [2.34.1](https://github.com/aws-amplify/amplify-swift/releases/tag/2.34.1)
